iphone/cli/commands/_build.js

```diff
@@ -2294,102 +2294,211 @@ iOSBuilder.prototype.scrubbedModuleId = function (moduleId) {
 
-	series(this, [
-		function (next) {
-			cli.emit('build.pre.construct', this, next);
-		},
+		// force the platform to "ios" just in case it was "iphone" so that plugins can reference it
+		cli.argv.platform = 'ios';
 
-		// initialization
-		'doAnalytics',
-		'initialize',
-		'determineLogServerPort',
-		'loginfo',
-		'readBuildManifest',
-		'checkIfNeedToRecompile',
-		'initBuildDir',
+		// Notify plugins that we're about to begin.
+		await new Promise((resolve, reject) => {
+			cli.emit('build.pre.construct', this, e => {
+				if (e) {
+					return reject(e);
+				}
+				resolve();
+			});
+		});
 
-		function (next) {
-			cli.emit('build.pre.compile', this, next);
-		},
+		// Post build anlytics.
+		await this.doAnalytics();
+
+		// Initialize build system. Checks if we need to do a clean or incremental build.
+		await this.initialize(); // sets a lot of fields/properties
+		await this.determineLogServerPort(); // needs to be run before loginfo and createXcodeProject (sets this.tiLogServerPort)
+		await this.loginfo(); // spits out fields to CLI
+		await this.readBuildManifest(); // sets this.previousBuildManifest and needs to be run before checkIfNeedToRecompile
+		await this.checkIfNeedToRecompile(); // compares various properties/fields versus last build
+		await this.initBuildDir(); // wipe or re-create build dir
+
+		// Notify plugins that we're prepping to compile.
+		await new Promise((resolve, reject) => {
+			cli.emit('build.pre.compile', this, e => {
+				if (e) {
+					return reject(e);
+				}
+				resolve();
+			});
+		});
 
-		function () {
-			// Make sure we have an app.js. This used to be validated in validate(), but since plugins like
-			// Alloy generate an app.js, it may not have existed during validate(), but should exist now
-			// that build.pre.compile was fired.
-			ti.validateAppJsExists(this.projectDir, this.logger, [ 'iphone', 'ios' ]);
-		},
+		// Make sure we have an "app.js" script. Will exit with a build failure if not found.
+		// Note: This used to be validated by the validate() method, but Alloy plugin
+		//       generates the "app.js" script via the "build.pre.compile" hook event above.
+		ti.validateAppJsExists(this.projectDir, this.logger, [ 'iphone', 'ios' ]);
 
 		// xcode related tasks
-		'createXcodeProject',
-		'writeEntitlementsPlist',
-		'writeInfoPlist',
-		'writeMain',
-		'copyTitaniumiOSFiles',
-		'copyExtensionFiles',
-		'cleanXcodeDerivedData',
+		// TODO: Can we use util.promisify(this.whatever).bind(this)()?
+		// Or just bite the bullet and move from callbacks to async impls for each...
+		await new Promise((resolve, reject) => {
+			this.createXcodeProject(e => {
+				if (e) {
+					return reject(e);
+				}
+				resolve();
+			});
+		});
+		await new Promise((resolve, reject) => {
+			this.writeEntitlementsPlist(e => {
+				if (e) {
+					return reject(e);
+				}
+				resolve();
+			});
+		});
+		this.writeInfoPlist();
+		await this.writeMain();
+		
```

Comment:
<!--
  1 failure:  iphone/cli/comman...
  0 warning: 
  
  
  DangerID: danger-id-Danger;
  File: iphone/cli/commands/_build.js;
  Line: 2361;
-->
- :no_entry_sign: iphone/cli/commands/_build.js line 2361 – Trailing spaces not allowed. (no-trailing-spaces)